File: app/src/main/java/workshop1024/com/xproject/controller/fragment/SavedFragment.java
package workshop1024.com.xproject.controller.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import workshop1024.com.xproject.R;

/**
 * 保存Fragment
 */
public class SavedFragment extends TopFragment {
    public SavedFragment() {
        // Required empty public constructor
    }

    public static SavedFragment newInstance() {
        SavedFragment fragment = new SavedFragment();
        fragment.setNavigationItemId(R.id.leftnavigator_menu_saved);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.saved_fragment, container, false);
    }
}

File: android/app/src/main/java/workshop1024/com/xproject/controller/fragment/save/SavedFragment.kt
package workshop1024.com.xproject.controller.fragment.save

import workshop1024.com.xproject.R
import workshop1024.com.xproject.controller.fragment.TopFragment
import workshop1024.com.xproject.controller.fragment.home.news.NewsListFragment

/**
 * 保存Fragment
 * //FIXME 和NewsListFragmet大量重复的代码，如何抽象，处理TopFragment和SubFragment的关系？？
 */
class SavedFragment : NewsListFragment(), TopFragment {

    override fun getNewsList() {
        mNewsRepository?.getSavedNewsList(this)
    }

    companion object {

        fun newInstance(): SavedFragment {
            val savedFragment = SavedFragment()
            savedFragment.mNavigationItemId = R.id.leftnavigator_menu_saved
            return savedFragment
        }
    }
}
